learned.lic

```diff
@@ -4,27 +4,105 @@
-  respond(output)
-end
+  def setup
+    arg_definitions = [
+      [
+        { name: 'combats', regex: /(combat|comb)/, optional: true, description: 'Show exp gained in combat skills.' },
+        { name: 'survival', regex: /(survival|surv)/, optional: true, description: 'Show exp gained in survival skills.' },
+        { name: 'magic', regex: /magic/, optional: true, description: 'Show exp gained in magic skills.' },
+        { name: 'armor', regex: /armor/, optional: true, description: 'Show exp gained in armor skills.' },
+        { name: 'weapons', regex: /(weapons|weap)/, optional: true, description: 'Show exp gained in weapon skills.' },
+        { name: 'lore', regex: /lore/, optional: true, description: 'Show exp gained in lore skills.' },
+        { name: 'guild', regex: /guild/, optional: true, description: 'Show exp gained in your guild skill.'}
+      ]
+    ]
+
+    args = parse_args(arg_definitions)
+
+    survival_skills = ["Evasion", "Athletics", "Perception", "Stealth", "Locksmithing", "Thievery", "First Aid", "Outdoorsmanship", "Skinning"]
+    lore_skills = ["Alchemy", "Appraisal", "Enchanting", "Forging", "Mechanical Lore", "Performance", "Scholarship", "Tactics"]
+    armor_skills = ["Shield Usage", "Chain Armor", "Plate Armor", "Light Armor", "Brigandine", "Defending"]
+    weapon_skills = ["Parry Ability", "Small Edged", "Larged Edged", "Twohanded Edged", "Small Blunt", "Large Blunt", "Twohanded Blunt", "Slings", "Bow", "Crossbow", "Staves", "Polearms", "Light Thrown", "Heavy Thrown", "Brawling", "Offhand Weapon", "Melee Mastery", "Missile Mastery"]
+    magic_skills = ["Arcane Magic", "Holy Magic", "Life Magic", "Elemental Magic", "Lunar Magic", "Attunement", "Arcana", "Targeted Magic", "Inner Fire", "Inner Magic", "Augmentation", "Debilitation", "Utility", "Warding", "Sorcery"]
+    guild_skills = ["Empathy", "Astrology", "Expertise", "Scouting", "Backstab", "Summoning", "Bardic Lore" "Conviction", "Theurgy", "Thanatology", "Trading"]
+
+    if args.survival
+      @skills_to_show.concat(survival_skills)
+    elsif args.combats
+      @skills_to_show.concat(armor_skills)
+                     .concat(weapon_skills)
+                     .push('Targeted Magic')
+                     .push('Debilitation')
+                     .push('Evasion')
+                     .push('Tactics')
+    elsif args.magic
+      @skills_to_show.concat(magic_skills)
+    elsif args.armor
+      @skills_to_show.concat(armor_skills)
+    elsif args.weapons
+      @skills_to_show.concat(weapon_skills)
+    elsif args.lore
+      @skills_to_show.concat(lore_skills)
+    elsif args.guild
+      @skills_to_show.concat(guild_skills)
+    else
+      @skills_to_show.concat(survival_skills)
+                     .concat(lore_skills)
+                     .concat(armor_skills)
+                     .concat(weapon_skills)
+                     .concat(magic_skills)
+                     .concat(guild_skills)
+    end
+  end
+
+  pause
```

Comment:
Why is this `pause` here?

Reply:
I am not sure, it was there initially so I left it in.